newIDE/app/src/MainFrame/index.js

```diff
@@ -449,6 +422,63 @@ const MainFrame = (props: Props) => {
     []
   );
 
+  React.useEffect(
+    () => {
+      if (isInitialProjectLoadingDone.current) return;
+      const autoLoadProject = async () => {
+        const {
+          getAutoOpenMostRecentProject,
+          getRecentProjectFiles,
+          hadProjectOpenedDuringLastSession,
+        } = preferences;
+
+        if (initialFileMetadataToOpen) {
+          // Open the initial file metadata (i.e: the file that was passed
+          // as argument and recognized by a storage provider). Note that the storage
+          // provider is assumed to be already set to the proper one.
+          const storageProviderOperations = getStorageProviderOperations();
+          const proceed = await ensureInteractionHappened(
+            storageProviderOperations
+          );
+          if (proceed) openInitialFileMetadata();
+        } else if (
+          getAutoOpenMostRecentProject() &&
+          hadProjectOpenedDuringLastSession() &&
+          getRecentProjectFiles()[0]
+        ) {
+          // Re-open the last opened project, if any and if asked to.
+          const fileMetadataAndStorageProviderName = getRecentProjectFiles()[0];
+          const storageProvider = findStorageProviderFor(
+            i18n,
+            props.storageProviders,
+            fileMetadataAndStorageProviderName
+          );
+          if (!storageProvider) return;
+          if (storageProvider.needUserAuthentication && loginState !== 'done')
+            return;
+
+          const storageProviderOperations = getStorageProviderOperations({
+            storageProvider,
+          });
+          const proceed = await ensureInteractionHappened(
+            storageProviderOperations
+          );
+          if (proceed)
+            openFromFileMetadataWithStorageProvider(
+              fileMetadataAndStorageProviderName
+            );
+        } else {
+          // Open the intro dialog if not opening any project.
+          if (introDialog) openIntroDialog(true);
+        }
+        isInitialProjectLoadingDone.current = true;
```

Comment:
Good question! I think it's because since everything happens at start and through effects, I wanted to make sure I have a variable that does not trigger re-renders

Reply:
alright 👍